pkg/ring/model.go

```diff
@@ -165,6 +165,9 @@ func (i *IngesterDesc) IsHealthy(op Operation, heartbeatTimeout time.Duration) b
 
 	case Ruler:
 		healthy = i.State == ACTIVE
+
+	case Compactor:
+		healthy = i.State == ACTIVE
```

Comment:
Slightly different than the previous `Read`, but this one should be more accurate.